src/uncrustify.cpp

```diff
@@ -2227,7 +2227,7 @@ void uncrustify_file(const file_mem &fm, FILE *pfout, const char *parsed_file,
       }
       newlines_insert_blank_lines();
 
-      if (options::pos_bool() != TP_IGNORE)
+      if (options::pos_bool() != TP_IGNORE || (options::code_width() > 0 && options::nl_bool_expression_multiline()))
```

Comment:
I like the idea, but it needs more work. `nl_bool_expression_multiline` cannot override `pos_bool`, it has to work with it.
The implementation in this PR is basically forcing a `trail` split on long lines. This may not be what the user expects based on the `pos_bool` setting. For example if `pos_bool` is `ignore`, the user would expect no changes at all. If `pos_bool` is `lead`, the user would expect the logical operator to be at the beginning of the next line, not at the end of the current long line.
`nl_bool_expression_multiline` should be active if `pos_bool` is different from `ignore`,  `join` or `break` and then it should place the logical operator based on the `pos_bool` value (`lead` or `trail`).

@guy-maurel what do you think about this?